cdap-api-spark-base/src/main/java/co/cask/cdap/api/spark/JavaSparkExecutionContextBase.java

```diff
@@ -113,6 +113,22 @@
    */
   public abstract MessagingContext getMessagingContext();
 
+  /**
+   * Returns a {@link MetadataReader} which can be used to read metadata. Currently the returned instance can
+   * only be used in the Spark driver process.
+   *
+   * @return A {@link MetadataReader}
+   */
+  public abstract MetadataReader getMetadataReader();
```

Comment:
Why only available for Java? Every spark API should be exposed through the Scala one as well.

Reply:
I have exposed them in SparkExecutionContextBase in earlier PRs.